File: mambo-protocol/src/main/scala/org/mambo/protocol/types/game/context/fight/GameFightTaxCollectorInformations.scala





















// Generated on 05/03/2013 10:35:04
// Scala version by Blackrush
package org.mambo.protocol.types

import org.mambo.core.io._
import org.mambo.protocol.{Type => BaseType, _}
import org.mambo.protocol.enums._

trait GameFightTaxCollectorInformations extends GameFightAIInformations {
  override val typeId = 48

  protected trait Type extends GameFightAIInformations.Type with Serializable {
    override val typeId = GameFightTaxCollectorInformations.typeId

    val firstNameId: Short
    val lastNameId: Short
    val level: Short

    override def serialize(buf: Buffer) {
      super.serialize(buf)

      buf.writeShort(firstNameId)
      buf.writeShort(lastNameId)
      buf.writeShort(level)
    }

  }

  protected def create(contextualId: Int, look: EntityLook.TypeImpl, disposition: EntityDispositionInformations.TypeImpl, teamId: Byte, alive: Boolean, stats: GameFightMinimalStats.TypeImpl, firstNameId: Short, lastNameId: Short, level: Short): Target = throw new IllegalAccessException

  override def deserialize(buf: Buffer) = {
    val contextualId = buf.readInt()
    val look = EntityLook.deserialize(buf)
    val disposition = types.get[EntityDispositionInformations.TypeImpl](buf.readShort());
    val teamId = buf.readByte()
    val alive = buf.readBoolean()
    val stats = types.get[GameFightMinimalStats.TypeImpl](buf.readShort());
    val firstNameId = buf.readShort()
    val lastNameId = buf.readShort()
    val level = buf.readShort()

    create(contextualId, look, disposition, teamId, alive, stats, firstNameId, lastNameId, level)
  }

}

object GameFightTaxCollectorInformations extends GameFightTaxCollectorInformations {
  case class TypeImpl(contextualId: Int, look: EntityLook.TypeImpl, disposition: EntityDispositionInformations.TypeImpl, teamId: Byte, alive: Boolean, stats: GameFightMinimalStats.TypeImpl, firstNameId: Short, lastNameId: Short, level: Short) extends Type
  override def create(contextualId: Int, look: EntityLook.TypeImpl, disposition: EntityDispositionInformations.TypeImpl, teamId: Byte, alive: Boolean, stats: GameFightMinimalStats.TypeImpl, firstNameId: Short, lastNameId: Short, level: Short) = TypeImpl(contextualId, look, disposition, teamId, alive, stats, firstNameId, lastNameId, level)

  def unapply(msg: TypeImpl) = TypeImpl.unapply(msg)
  def apply(contextualId: Int, look: EntityLook.TypeImpl, disposition: EntityDispositionInformations.TypeImpl, teamId: Byte, alive: Boolean, stats: GameFightMinimalStats.TypeImpl, firstNameId: Short, lastNameId: Short, level: Short) = create(contextualId, look, disposition, teamId, alive, stats, firstNameId, lastNameId, level)

  type Target = TypeImpl

}

File: mambo-protocol/src/main/java/org/mambo/protocol/types/GameFightTaxCollectorInformations.java


// Generated on 05/08/2013 19:38:01
package org.mambo.protocol.types;

import java.util.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class GameFightTaxCollectorInformations extends GameFightAIInformations {
    public static final short TYPE_ID = 48;
    
    @Override
    public short getTypeId() {
        return TYPE_ID;
    }
    
    public short firstNameId;
    public short lastNameId;
    public short level;
    
    public GameFightTaxCollectorInformations() { }
    
    public GameFightTaxCollectorInformations(int contextualId, EntityLook look, EntityDispositionInformations disposition, byte teamId, boolean alive, GameFightMinimalStats stats, short firstNameId, short lastNameId, short level) {
        super(contextualId, look, disposition, teamId, alive, stats);
        this.firstNameId = firstNameId;
        this.lastNameId = lastNameId;
        this.level = level;
    }
    
    @Override
    public void serialize(Buffer buf) {
        super.serialize(buf);
        buf.writeShort(firstNameId);
        buf.writeShort(lastNameId);
        buf.writeShort(level);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        super.deserialize(buf);
        firstNameId = buf.readShort();
        if (firstNameId < 0)
            throw new RuntimeException("Forbidden value on firstNameId = " + firstNameId + ", it doesn't respect the following condition : firstNameId < 0");
        lastNameId = buf.readShort();
        if (lastNameId < 0)
            throw new RuntimeException("Forbidden value on lastNameId = " + lastNameId + ", it doesn't respect the following condition : lastNameId < 0");
        level = buf.readShort();
        if (level < 0)
            throw new RuntimeException("Forbidden value on level = " + level + ", it doesn't respect the following condition : level < 0");
    }
    
}
